tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -600,14 +615,21 @@ static unique_ptr<TableFunctionRef> ScanReplacement(ClientContext &context, cons
 }
 
 shared_ptr<DuckDBPyConnection> DuckDBPyConnection::Connect(const string &database, bool read_only,
-                                                           const py::dict &config_dict) {
+                                                           py::object config_options) {
```

Comment:
This should remain of type `py::dict`

Reply:
py::dict is not a fan of my passing py::none() as default to it
I thought we wanted to remove all mutable objects as default argument, or did I miss interpret the issue?

Reply:
```py
tools/pythonpkg/tests/fast/test_insert.py:9: in TestInsert
    conn = duckdb.connect()
E   TypeError: connect(): incompatible function arguments. The following argument types are supported:
E       1. (database: str = ':memory:', read_only: bool = False, config: dict = None) -> duckdb.DuckDBPyConnection
E   
E   Invoked with:
```

Reply:
Yes, that error message is the result (to my understanding) of us not using `std::optional`

We should be using the `py::dict` as the type of the arg, it's the value of the default that should not be mutable

Reply:
Yes, that is what I mentioned in our meeting a couple of weeks ago. I don't think adding a single file header library is a huge thing?